### PR TITLE
Integrate 'DBus services exposure' of Atsec into openQA

### PR DESCRIPTION
**Pull request by Amrysliu**
Related: https://progress.opensuse.org/issues/109542
Verify run: https://openqa.suse.de/tests/8575490# (aarch64)
                 https://openqa.suse.de/tests/8575461# (64bit)
                 https://openqa.suse.de/tests/8575462# (s390x)

### CODE REVIEW

**Review by rfan1 — APPROVED** [review]

File: tests/security/atsec/dbus_fuzzer.pm [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    select_console 'root-console';
+
+    # Install the required packages
+    zypper_call('in glib2-devel libffi-devel gcc make');
+
+    # Compile
+    assert_script_run("cd $atsec_test::code_dir/pentest/dfuzzer-master/src");
+    assert_script_run('make');
+
+    my $output = script_output('./dfuzzer -l 2>&1');
```
Comment (rfan1) [inline tests/security/atsec/dbus_fuzzer.pm:31]: It is used for record some info here, correct?

File: tests/security/atsec/dbus_fuzzer.pm [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    my @unknown_bus_name = grep { !$hash_white_list{$_} } (@bus_list);
+
+    # After filtering there should be no unknown name.
+    if (scalar(@unknown_bus_name) > 0) {
+        record_info('There are unknow bus name', Dumper(\@unknown_bus_name), result => 'fail');
+        $self->result('fail');
+    }
+
+    # Create a directory to restore the log message generated by dfuzzer
+    my $log_dir = 'logs';
```
Comment (rfan1) [inline tests/security/atsec/dbus_fuzzer.pm:63]: Minor comments, use more complex dir name

**Review by rfan1 — APPROVED** [review]
LGTM

File: lib/atsec_test.pm [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Base module for audit-test test cases
```
Comment (lilyeyes) [inline lib/atsec_test.pm:6]: It might be ATSEC/atsec not audit, right?

File: tests/security/atsec/dbus_fuzzer.pm [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        }
+    }
+
+    record_info('Result of dfuzzer -l', Dumper(\@bus_list));
+
+    # Analyse the results
+    my %hash_white_list = map { $_ => 1 } @atsec_test::white_list_for_dbus;
+    my @unknown_bus_name = grep { !$hash_white_list{$_} } (@bus_list);
+
+    # After filtering there should be no unknown name.
```
Comment (lilyeyes) [inline tests/security/atsec/dbus_fuzzer.pm:56]: Delete the `.` at the end of line "56".

File: tests/security/atsec/dbus_fuzzer.pm (hunk 1 above)
Comment (Amrysliu) [reply tests/security/atsec/dbus_fuzzer.pm:31]: Not only for record, this is also a test point.

File: tests/security/atsec/dbus_fuzzer.pm (hunk 2 above)
Comment (Amrysliu) [reply tests/security/atsec/dbus_fuzzer.pm:63]: This directory is under the test directory, I think simple name is ok.

File: lib/atsec_test.pm (hunk 3 above)
Comment (Amrysliu) [reply lib/atsec_test.pm]: fixed

**Comment by lilyeyes** [comment]
For case "`4.3 DBus fuzzer`" in confluence page it says "This list is incorporated in `dfuzz.sh` file which is to be executed."
This `dfuzz.sh` does not be found in you code, do you have any replacement code for it?

Generally look good to me.

**Comment by Amrysliu** [comment]
> 

No, I didn't find this script. From the logs message, I think `dfuzz.sh` is used to testing every dbus name by running `./dfuzzer -v -n xxxx` and generate the log files. I did that in my test code, so I guess we don't need `dfuzz.sh`

**Comment by lilyeyes** [comment]
> @lilyeyes No, I didn't find this script. From the logs message, I think `dfuzz.sh` is used to testing every dbus name by running `./dfuzzer -v -n xxxx` and generate the log files. I did that in my test code, so I guess we don't need `dfuzz.sh`

Thanks.

**Review by lilyeyes — APPROVED** [review]
Thanks for the revise and explanation, LGTM now.

File: tests/security/atsec/dbus_fuzzer.pm [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    my ($self) = shift;
+
+    select_console 'root-console';
+
+    # Install the required packages
+    zypper_call('in glib2-devel libffi-devel gcc make');
+
+    # Compile
+    assert_script_run("cd $atsec_test::code_dir/pentest/dfuzzer-master/src");
+    assert_script_run('make');
```
Comment (evverx) [inline tests/security/atsec/dbus_fuzzer.pm:29]: `dfuzzer` kind of switched to `meson` in https://github.com/matusmarhefka/dfuzzer/pull/26 and can be built and run with
```
meson --buildtype=release build
ninja -C ./build
./build/dfuzzer ...
```
`Makefile` will be removed once the `dfuzzer` package switches to `meson` as well. Also it no longer depends on `libffi-devel`.

Comment (evverx) [inline tests/security/atsec/dbus_fuzzer.pm:29]: `dfuzzer` kind of switched to `meson` in https://github.com/matusmarhefka/dfuzzer/pull/26 and can be built and run with
```
meson --buildtype=release build
ninja -C ./build
./build/dfuzzer ...
```
`Makefile` will be removed once the `dfuzzer` package switches to `meson` as well. Also it no longer depends on `libffi-devel`.

File: tests/security/atsec/dbus_fuzzer.pm (hunk 1 above)
Comment (evverx) [inline tests/security/atsec/dbus_fuzzer.pm:31]: The output of `dfuzzer -l` changed recently and I suspect it broke the script. I opened https://github.com/matusmarhefka/dfuzzer/issues/67 to figure out what can be done to make it easier to use it in scripts like this.

**Pull request merged**